lib/ansible/modules/system/user.py

```diff
@@ -532,17 +535,22 @@ def modify_user_usermod(self):
             cmd.append('-s')
             cmd.append(self.shell)
 
-        if self.expires:
+        if self.expires is not None:
+
             current_expires = self.user_password()[1]
 
-            # Convert days since Epoch to seconds since Epoch as struct_time
-            total_seconds = int(current_expires) * 86400
-            current_expires = time.gmtime(total_seconds)
+            if self.expires < time.gmtime(0):
+                if current_expires >= 0:
```

Comment:
This should be `> 0`. Having it set to `>= 0` causes a change to always be made even if the current expiration is 0.

Reply:
fixed